mlir/lib/Dialect/OpenACC/IR/OpenACC.cpp

```diff
@@ -2748,6 +2748,49 @@ void acc::LoopOp::addEmptyWorker(
                                                    effectiveDeviceTypes));
 }
 
+void acc::LoopOp::addEmptyGang(
+    MLIRContext *context, llvm::ArrayRef<DeviceType> effectiveDeviceTypes) {
+  setGangAttr(addDeviceTypeAffectedOperandHelper(context, getGangAttr(),
+                                                 effectiveDeviceTypes));
+}
+
+void acc::LoopOp::addGangOperands(
+    MLIRContext *context, llvm::ArrayRef<DeviceType> effectiveDeviceTypes,
+    llvm::ArrayRef<GangArgType> argTypes, mlir::ValueRange values) {
+  llvm::SmallVector<int32_t> segments;
+  if (getGangOperandsSegments())
```

Comment:
Can you save the return value here so you don't have to make this call again on the next line?